### ISSUE TITLE
HTML email's text representation includes too many linebreaks

### ISSUE THREAD

**Issue by foobear** [makandra/spreewald#153]
`MailFinder.email_text_body` deduplicates multiple `\n` newlines before it converts `\r\n` linebreaks to `\n`. That way, newlines in bodies with only `\r\n` linebreaks are actually never deduplicated.

For example,

```ruby
"<p>\r\nHello!\r\n</p>\r\n<p>\r\nBye!\r\n</p>"
```

results in

```ruby
"Hello!\n\n\nBye!"
```

Relevant code:

https://github.com/makandra/spreewald/blob/db95ff30ba079d9e157ee50f3b054c172d7a1063/lib/spreewald_support/mail_finder.rb#L60-L70

While we could just move `gsub(/\n\n/, "\n")` from the `html_part` section to the end of `email_text_body` and always deduplicate, this would impact plain-text bodies negatively.
Maybe use `gsub(/\r?\n\r?\n/, "\n")` in the `html_part` section instead. Or `gsub(/\r?\n(?:\r?\n)/, "\n")`, maybe?

Please also provide a 2.99.x backport for this to allow easier migration to 3.0.

**Comment by kratob** [comment]
I suggest using `[\r\n]+` for the regexp.

**Comment by foobear** [comment]
Note that this would discard blank lines in plain-text bodies, and tests expecting those would fail.

Putting `gsub(/[\r\n]+/, "\n")` only in the HTML part's branch should be fine.
Alternatively, `gsub(/[\r\n]{2,}/, "\n")` could be an option.

**Comment by FLeinzi** [comment]
Yes, @kratob means the regexp in line 63

**Comment by FLeinzi** [comment]
I have fixed this in `v3.0.1`.

There is also a backport in `v2.99.2`
